Content.Server/Body/Systems/BodySystem.cs

```diff
@@ -136,22 +130,17 @@ public override HashSet<EntityUid> GibBody(EntityUid bodyId, bool gibOrgans = fa
 
         _audio.Play(body.GibSound, filter, coordinates, true, audio);
 
-        var containers = GetBodyContainers(bodyId, body: body).ToList();
-
-        foreach (var container in containers)
+        foreach (var entity in gibs)
         {
-            foreach (var entity in container.ContainedEntities)
+            if (deleteItems)
+            {
+                QueueDel(entity);
+            }
+            else
             {
-                if (deleteItems)
-                {
-                    QueueDel(entity);
-                }
-                else
-                {
-                    container.Remove(entity, EntityManager, force: true);
-                    SharedTransform.SetCoordinates(entity,coordinates);
-                    entity.RandomOffset(0.25f);
-                }
+                var random = IoCManager.Resolve<IRobustRandom>();
+                var offset = new Vector2(-.25f + random.NextFloat() * .5f, -.25f + random.NextFloat() * .5f);
```

Comment:
you can probably do this with the NextVector2 method on IRobustRandom. You can optionally specify min and max values, which seems to be what you're doing here.

Reply:
Done.